topic-controller/src/main/java/io/strimzi/controller/topic/TopicWatcher.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.strimzi.controller.topic;
+
+import io.strimzi.controller.topic.zk.Zk;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * ZooKeeper watcher for child znodes of {@code /brokers/topics},
+ */
+public class TopicWatcher {
+
+    private static final Logger log = LoggerFactory.getLogger(TopicWatcher.class);
+
+    private static final String TOPICS_ZNODE = "/brokers/topics";
+
+    private final Controller controller;
+
+    private final ConcurrentHashMap<String,Boolean> children = new ConcurrentHashMap<>();
+    private volatile int state = 0;
+    private volatile Zk zk;
+
+    /**
+     * Contructor
+     *
+     * @param controller    Controller instance to notify
+     */
+    TopicWatcher(Controller controller) {
+        this.controller = controller;
+    }
+
+    /**
+     * Add a child to watch under the topics znode
+     *
+     * @param child child to watch
+     */
+    public void addChild(String child) {
+        log.debug("Watching topic {} for partitions changes", child);
+        this.children.put(child, false);
```

Comment:
You switched from using `Boolean.FALSE`, which means the primitive has to be boxed (but maybe the JIT compiler is clever enough to figure this out these days). I guess primitives is more readable though, and boxing is unlikely to make any difference here, so let's leave keep it like this.